File: client/src/config/permissionMap.js
// src/config/permissionMap.js

/**
 * 前端权限映射配置
 * 将页面、菜单、按钮映射到所需的权限
 */

// 菜单权限映射
export const MENU_PERMISSION_MAP = {
  'menu:leak:scan': ['leak:view'],
  'menu:assessment': ['assessment:view'],
  'menu:system:settings': ['user:view', 'role:view', 'permission:view'],
};

// 页面权限映射
export const PAGE_PERMISSION_MAP = {
  '/leak-scan': ['leak:view'],
  '/assessment/projects': ['assessment:view'],
  '/assessment/rules': ['assessment:view'],
  '/system/users': ['user:view'],
  '/system/roles': ['role:view'],
  '/system/permissions': ['permission:view'],
};

// 按钮权限映射
export const BUTTON_PERMISSION_MAP = {
  // 用户管理按钮
  'button:user:create': ['user:create'],
  'button:user:edit': ['user:edit'],
  'button:user:delete': ['user:delete'],
  'button:user:assign_role': ['user:edit'],
  
  // 角色管理按钮
  'button:role:create': ['role:manage'],
  'button:role:edit': ['role:manage'],
  'button:role:delete': ['role:manage'],
  'button:role:assign_permission': ['role:manage'],
  
  // 漏扫处理按钮
  'button:leak:extract': ['leak:extract'],
  'button:leak:export': ['leak:export'],
  
  // 测评录入按钮
  'button:assessment:create': ['assessment:manage'],
  'button:assessment:edit': ['assessment:manage'],
  'button:assessment:delete': ['assessment:manage'],
};

/**
 * 检查是否有权限访问菜单
 */
export const canAccessMenu = (userPermissions, menuCode) => {
  const requiredPerms = MENU_PERMISSION_MAP[menuCode];
  if (!requiredPerms) return true;
  return requiredPerms.some(perm => userPermissions.includes(perm));
};

/**
 * 检查是否有权限访问页面
 */
export const canAccessPage = (userPermissions, path) => {
  const requiredPerms = PAGE_PERMISSION_MAP[path];
  if (!requiredPerms) return true;
  return requiredPerms.some(perm => userPermissions.includes(perm));
};

/**
 * 检查是否有权限使用按钮
 */
export const canUseButton = (userPermissions, buttonCode) => {
  const requiredPerms = BUTTON_PERMISSION_MAP[buttonCode];
  if (!requiredPerms) return true;
  return requiredPerms.some(perm => userPermissions.includes(perm));
};

File: server/app/permission_map.py
# -*- coding: utf-8 -*-

"""
API权限映射配置
将API接口映射到所需的权限
"""

# API权限映射表
# 格式: 'METHOD:/api/path': ['permission1', 'permission2']
API_PERMISSION_MAP = {
    # ========== 认证相关 ==========
    'POST:/api/auth/login': [],  # 登录不需要权限
    'GET:/api/auth/current-user': ['user:view'],
    'GET:/api/auth/user-permissions': ['user:view'],
    
    # ========== 用户管理 ==========
    'GET:/api/users': ['user:view'],
    'GET:/api/users/*': ['user:view'],
    'POST:/api/users': ['user:create'],
    'PUT:/api/users/*': ['user:edit'],
    'DELETE:/api/users/*': ['user:delete'],
    'POST:/api/users/*/roles': ['user:edit'],
    'DELETE:/api/users/*/roles/*': ['user:edit'],
    
    # ========== 角色管理 ==========
    'GET:/api/roles': ['role:view'],
    'GET:/api/roles/*': ['role:view'],
    'POST:/api/roles': ['role:manage'],
    'PUT:/api/roles/*': ['role:manage'],
    'DELETE:/api/roles/*': ['role:manage'],
    'POST:/api/roles/*/permissions': ['role:manage'],
    'DELETE:/api/roles/*/permissions/*': ['role:manage'],
    
    # ========== 权限管理 ==========
    'GET:/api/permissions': ['permission:view'],
    'GET:/api/permissions/*': ['permission:view'],
    'GET:/api/permissions/resources': ['permission:view'],
    
    # ========== 漏扫处理 ==========
    'GET:/api/leak-scan/history': ['leak:view'],
    'POST:/api/leak-scan/extract': ['leak:extract'],
    'POST:/api/leak-scan/link-to-project': ['leak:export'],
    
    # ========== 测评录入 ==========
    'GET:/api/projects': ['assessment:view'],
    'GET:/api/projects/*': ['assessment:view'],
    'POST:/api/projects': ['assessment:manage'],
    'PUT:/api/projects/*': ['assessment:manage'],
    'DELETE:/api/projects/*': ['assessment:manage'],
    'GET:/api/rules': ['assessment:view'],
    'GET:/api/rules/*': ['assessment:view'],
    'POST:/api/rules': ['assessment:manage'],
    'PUT:/api/rules/*': ['assessment:manage'],
    'DELETE:/api/rules/*': ['assessment:manage'],
}


def get_required_permissions(method, path):
    """
    获取API接口所需的权限列表
    
    Args:
        method: HTTP方法 (GET, POST, PUT, DELETE)
        path: 请求路径
    
    Returns:
        list: 所需的权限列表
    """
    # 精确匹配
    exact_key = f"{method}:{path}"
    if exact_key in API_PERMISSION_MAP:
        return API_PERMISSION_MAP[exact_key]
    
    # 通配符匹配
    for key, permissions in API_PERMISSION_MAP.items():
        if '*' in key:
            # 提取匹配模式
            key_method, key_path = key.split(':', 1)
            if key_method != method:
                continue
            
            # 将通配符转换为正则表达式
            import re
            pattern = key_path.replace('*', '[^/]+')
            if re.match(f"^{pattern}$", path):
                return permissions
    
    # 默认需要认证（但不需要特定权限）
    return ['authenticated']